src/simulator_core/entities/assets/esdl_asset_object.py

```diff
@@ -66,30 +38,14 @@ def __init__(self, asset: esdl.Asset) -> None:
         :param asset, esdl.Asset: PyEsdl Asset object
         """
         self.esdl_asset = asset
-        self.asset_specific_parameters = self.get_property_dict_for_asset()
-
-    def get_property_dict_for_asset(self) -> dict:
-        """
-        Get the properties of the ESDL asset required for the simulation.
 
-        :return: dict of properties
-        """
-        # Retrieve the entinty type of the asset
-        asset_type_string = StringEsdlAssetMapper().to_entity(type(self.esdl_asset))
-        return ASSET_DICT[asset_type_string]
-
-    def get_asset_parameters(self) -> dict:
-        """
-        Get the parameters of the asset.
+    def get_property(self, esdl_property_name: str, default_value: Any) -> (Any, bool):
+        """Get property value from the esdl_asset based on the "ESDL" name.
 
-        :return: dict of parameters
+        :return: Tuple with the value of the property and a boolean indicating whether the property
+        was found in the esdl_asset.
         """
-        parameters = {}
-        for parameter_key, esdl_key in self.asset_specific_parameters.items():
-            try:
-                parameters[parameter_key] = getattr(self.esdl_asset, esdl_key.name)
-            except AttributeError:
-                logger.warning(f"Attribute {esdl_key.name} not found in {self.esdl_asset}.")
-                logger.warning(f"Default value of {esdl_key.default} used for {esdl_key.name}.")
-                parameters[parameter_key] = esdl_key.default
-        return parameters
+        try:
+            return (getattr(self.esdl_asset, esdl_property_name), True)
```

Comment:
je zou dit ook kunnen doen met  de hasattr function. ik weet even niet wat voor en nadelen vna beide optie zijn

Reply:
https://stackoverflow.com/a/24971134/15619397 hasattr only when you don't need the value